### ISSUE TITLE
Debian Packaging Incompatibility with OpenJDK

### ISSUE THREAD

**Issue by david-x64** [adoptium/installer#1077]
### Please provide a brief summary of the bug

Debian package `temurin-21-jdk` provides package name `java21-sdk-headless`.
But it does not provide `java21-sdk`.

Debian package `openjdk-21-jdk` on linux mint provides `java21-sdk`
But it does not provide `java21-sdk-headless`.

So there is no common package name provided anymore which I can use to express a build dependency on Java 21 in a vendor independent manner.

Interestingly, the `temurin-21-jdk` package does provide the `java<version>-sdk` package name for Java versions up to 18.

### Did you test with the latest update version?

- [X] Yes

### Please provide steps to reproduce where possible

Compare package descriptions of `temurin-21-jdk` and `openjdk-21-jdk`.
Especially compare the field `Provides` of both packages.
```
apt-cache show temurin-21-jdk
apt-cache show openjdk-21-jdk
```

### Expected Results

I expect to find a versioned common name in the `Provides` field of both packages.
For example `temurin-21-jdk` should provide `java21-sdk` just as `openjdk-21-jdk` does already.

### Actual Results

`temurin-21-jdk` does not provide `java21-sdk`.

### What Java Version are you using?

openjdk 21.0.5 2024-10-15 LTS OpenJDK Runtime Environment Temurin-21.0.5+11 (build 21.0.5+11-LTS) OpenJDK 64-Bit Server VM Temurin-21.0.5+11 (build 21.0.5+11-LTS, mixed mode, sharing)

### What is your operating system and platform?

Linux Mint 22 - Wilma, x86_64
```
NAME="Linux Mint"
VERSION="22 (Wilma)"
ID=linuxmint
ID_LIKE="ubuntu debian"
PRETTY_NAME="Linux Mint 22"
VERSION_ID="22"
HOME_URL="https://www.linuxmint.com/"
SUPPORT_URL="https://forums.linuxmint.com/"
BUG_REPORT_URL="http://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/"
PRIVACY_POLICY_URL="https://www.linuxmint.com/"
VERSION_CODENAME=wilma
UBUNTU_CODENAME=noble
```

### How did you install Java?

```
sudo apt install temurin-21-jdk
```

### Did it work before?

```Shell
Yes, this worked up to Java-18.
```

### Relevant log output

```Shell
Package: temurin-21-jdk
Version: 21.0.0.0.0+35
...
Provides: ...java-sdk, java-sdk-headless, ..., java18-sdk, java21-sdk-headless, ...
```
```

**Comment by steelhead31** [comment]
This may be an omission in the debian control files used to build the .deb packages, sample below, it looks like after JDK18, some updates were omitted, I guess determining if this was deliberate or not should influence the fix.   
  
         `java-sdk,`  
 `java-sdk-headless,`  
 `java10-sdk,`  
 `java11-sdk,`  
 `java12-sdk,`  
 `java13-sdk,`  
 `java14-sdk,`  
 `java15-sdk,`  
 `java16-sdk,`  
 `java17-sdk,`  
 `java18-sdk,`  
 `java2-sdk,`  
 `java5-sdk,`  
 `java6-sdk,`  
 `java7-sdk,`  
 `java8-sdk,`  
 `java9-sdk,`  
 `java10-sdk-headless,`  
 `java11-sdk-headless,`  
 `java12-sdk-headless,`  
 `java13-sdk-headless,`  
 `java14-sdk-headless,`  
 `java15-sdk-headless,`  
 `java16-sdk-headless,`  
 `java17-sdk-headless,`  
 `java18-sdk-headless,`  
 `java19-sdk-headless,`  
 `java2-sdk-headless,`  
 `java20-sdk-headless,`  
 `java21-sdk-headless,`  
 `java5-sdk-headless,`  
 `java6-sdk-headless,`  
 `java7-sdk-headless,`  
 `java8-sdk-headless,`  
 `java9-sdk-headless,`  
 `java-runtime (= 21),`  
 `java-runtime-headless (= 21),`  
 `java10-runtime,`  
 `java11-runtime,`  
 `java12-runtime,`  
 `java13-runtime,`  
 `java14-runtime,`  
 `java15-runtime,`  
 `java16-runtime,`  
 `java17-runtime,`  
 `java18-runtime,`  
 `java19-runtime,`  
 `java2-runtime,`  
 `java20-runtime,`  
 `java21-runtime,`

**Comment by steelhead31** [comment]
After discussion at PMC, this issue appears to be a bug/omission when the debian control files were created for JDK21 onwards, this will be fixed in the upcoming January 2025 CPU cycle. A new issue will be created to discuss whether JDK21 should be providing the earlier runtime/sdk/etc version, or behave in a similar fashion to the rpms.

**Comment by steelhead31** [comment]
New issue to track future enhancement/potential change : #1096

**Comment by steelhead31** [comment]
Also interestingly the control files used to build the jre deb packages do have the lines I believe are missing for the JDK.

**Comment by sxa** [comment]
> Also interestingly the control files used to build the jre deb packages do have the lines I believe are missing for the JDK.

Sounds even more like just a simple oversight then :-)

**Comment by steelhead31** [comment]
This commit, made as part of the installer refactor has the required fixes.. f4f1468    I've also made the changes to the existing package creation mechanism, should those be required once the new automated process is in place.

**Comment by steelhead31** [comment]
All fixed, and new packages published for the January 2025 release cycle.
